nat-lab/tests/conftest.py

```diff
@@ -196,6 +202,52 @@ def pytest_make_parametrize_id(config, val):
     return param_id
 
 
+async def setup_check_connectivity():
+    if "GITLAB_CI" not in os.environ:
+        return
+
+    test_nodes = {
+        ConnectionTag.WINDOWS_VM_1: [
+            LAN_ADDR_MAP[ConnectionTag.WINDOWS_VM_2],
+            LAN_ADDR_MAP[ConnectionTag.MAC_VM],
+            DERP_PRIMARY.ipv4,
+        ],
+        ConnectionTag.WINDOWS_VM_2: [
+            LAN_ADDR_MAP[ConnectionTag.WINDOWS_VM_1],
+            LAN_ADDR_MAP[ConnectionTag.MAC_VM],
+            DERP_PRIMARY.ipv4,
+        ],
+        ConnectionTag.MAC_VM: [
+            LAN_ADDR_MAP[ConnectionTag.WINDOWS_VM_1],
+            LAN_ADDR_MAP[ConnectionTag.WINDOWS_VM_2],
+            DERP_PRIMARY.ipv4,
+        ],
+    }
+    results: Dict[ConnectionTag, List[Tuple[str, bool]]] = {
+        key: [] for key in test_nodes
+    }
+    for source, destinations in test_nodes.items():
+        for dest_ip in destinations:
+            try:
+                async with new_connection_with_conn_tracker(source, None) as (
+                    connection,
+                    _,
+                ):
+                    await ping(connection, dest_ip, 5)
+                    results[source].append((dest_ip, True))
```

Comment:
Very minor: can we do reverse ip lookup and insert hostname instead in here to avoid having map the IP to the machine?